src/Fmt_ast.ml

```diff
@@ -3806,9 +3806,14 @@ and fmt_structure_item c ~last:last_item ?ext {ctx; ast= si} =
 
 and fmt_let c ctx ~ext ~rec_flag ~bindings ~parens ~attributes ~fmt_atrs
     ~fmt_expr ~indent_after_in =
+  let fmt_in indent =
+    match c.conf.break_before_in with
+    | `Always -> break 1 (-indent) $ fmt "in"
+    | `Auto -> fits_breaks " in" ("@;<1 " ^ Int.to_string (-indent) ^ ">in")
```

Comment:
I don't like this. The first thing `fits_breaks` does is parse its second argument, that is not a format bug looks like one. This is unnecessary code that adds confusion.

I suggest adding `fits_breaks'` function that takes the break values as additional arguments for when they are variable.

Reply:
I'm in favor of modifying `Fmt.fits_breaks` instead of having 2 functions:
```ocaml
val fits_breaks :
     ?force_fit_if:bool
  -> ?force_break_if:bool
  -> ?nspaces:int
  -> ?offset:int
  -> string
  -> string
  -> t
(** [fits_breaks fits nspaces offset breaks] prints [fits] if the enclosing
    box fits on one line, and otherwise prints [breaks], which is a string
    that optionally follows a break hint equivalent to
    ["@;<nspaces offset>"]. *)
```
it is in conflict with #311  but I'm really not sure the new feature proposed in #311 will really be used in ocamlformat (I did a quick check in Fmt_ast).

Reply:
This must be fixed but could be in an other PR. This modified `fits_breaks` looks good.